python_benchmarks/global_params.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import torch
+
+# Correctness Tests tolerance
+RTOL = 1e-3
+ATOL = 1e-3
+
+
+# Common parameters for all tests
+def generate_input_sizes():
+    range_outer = [2**i for i in range(1, 5)]
+    range_inner = [32 * 1024 * 2**i for i in range(11)]
+
+    inputs = [(i, j) for i in range_outer for j in range_inner]
+    inputs.extend([(j, i) for i in range_outer for j in range_inner])
+    return inputs
+
+
+INPUT_SIZES = generate_input_sizes()
```

Comment:
This assumes the input is 2D? That's a bit too big assumption to put in a **global** parameter. I'd provide helper methods to generate input in certain dimensions and let benchmarks call these explicitly.

Reply:
These 2D inputs are used by a majority of benchmarks so it greatly simplifies their definition. If we remove this to have the benchmarks explicitly use helper functions, we will have to parametrize inputs for all the benchmark functions. I would prefer directly parametrizing for only the set of benchmarks that require another set of inputs (we can have helper functions for that) and not import pytestmark in those cases.

Reply:
Switched to explicit parametrization for each benchmark.